### ISSUE TITLE
Updating downloaded items

### ISSUE THREAD

**Issue by stevenbird** [aikuma/aikuma#371]
Cloud-sync should update the items that have already been downloaded, if new commentaries have been provided. (This relates to #243.)

**Comment by stevenbird** [comment]
If there's been a lot of new commentaries, it is probably not the right thing to download them all. Instead, the user should just be alerted to the fact that there is new material. Either the server needs to track each client device to know which items should be updated (alerted, pushed), or the client must query the server for any updates since the last update time.

**Comment by stevenbird** [comment]
In order for the client to do the work, it must submit a query for updates concerning any of the items it currently has (which could be thousands). The [fusion table language](https://developers.google.com/fusiontables/docs/v1/sql-reference) does not support disjunction but it does allow `<column_name> IN <string1>, <string2>, ...`, so we would only be limited by the allowable length of a query. Alternatively, the `FusionIndex2` interface could do the work of breaking call down into several queries, but then we need to be aware of the limit of [25k reads per day](https://developers.google.com/fusiontables/docs/v1/using#quota).

**Comment by stevenbird** [comment]
The expected size limits are:
* HTTP: [8192 bytes](http://stackoverflow.com/questions/2659952/maximum-length-of-http-get-request) (12 bytes per item_id = about 600 items in a single HTTP GET request)
* Fusion Table Query Language: [unspecified](https://developers.google.com/fusiontables/docs/v2/getting_started) (could we fit 600 items in a query?)

**Comment by stevenbird** [comment]
We will assume that the length limit for a fusion table query is less than the HTTP size limit, so only the client has to break it into multiple queries.

**Comment by stevenbird** [comment]
This is blocked on #380.

**Comment by stevenbird** [comment]
For now, this sync will be done manually, when the user selects the `GoogleCloud Sync` option on the continuation menu.

**Comment by stevenbird** [comment]
We need to decide between two options:
1. the client queries the server for any items that have been updated and downloads them
2. the server tracks which items have been "downloaded" by the user (to their Google Drive, thence synced to the phone), and shares any new files for those items (and these require no special sync any more, since it can be covered by the backup functionality which will restore a user's files back to the client, cf #232)

**Comment by stevenbird** [comment]
For option 2, we need the following: (i) a service which receives a request from the client to share an item; (ii) tracking any items that the user is "following" (client uses the FusionIndex to log the fact that this item is shared)

**Comment by stevenbird** [comment]
@hleeldc would you please confirm that we're going with option 2?

**Comment by stevenbird** [comment]
This issue is superseded by #431 and #434.
